### PR TITLE
✨Source Coda: Migrate Python CDK to Low-code CDK

### LINKED ISSUE #29264: Coda
https://docs.airbyte.com/integrations/sources/coda

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
*Migrating Source Coda to Low-Code CDK*
Closes #29264

## How
*Developed using (Configuration Based Source) low-code CDK*

## Recommended reading order
1. `spec.yaml`
2. `manifest.yaml`
3. `schemas/*`

## 🚨 User Impact 🚨
- No breaking changes, migration to low-code
<details><summary><strong>Migrate Connector: Source Coda</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

</details>

### CODE REVIEW

**Comment by btkcodedev** [comment]
Test result - passed state

image

**Comment by btkcodedev** [comment]
Checks:
- Added paginator
- No incremental sync support
- Updated version

All test passing
image

**Review by sh4sh — APPROVED** [review]

**Review by sh4sh — CHANGES_REQUESTED** [review]
We're missing the `rows` stream. Once this is added, we can merge :) https://github.com/airbytehq/airbyte/blob/7b85af0244b69a2bb1f97849cddbf3e2cb332458/airbyte-integrations/connectors/source-coda/source_coda/source.py#L117-L130

File: airbyte-integrations/connectors/source-coda/acceptance-test-config.yml [hunk 1]
```diff
@@ -24,16 +22,14 @@ acceptance_tests:
...
-  #          path: "integration_tests/expected_records.jsonl"
-  #          extra_fields: no
-  #          exact_order: no
-  #          extra_records: yes
-  incremental:
+            bypass_reason: "Sandbox account can't seed the stream"
+          - name: controls
+            bypass_reason: "Sandbox account can't seed the stream"
+          - name: tables
+            bypass_reason: "Sandbox account can't seed the stream"
```
Comment (sh4sh) [inline airbyte-integrations/connectors/source-coda/acceptance-test-config.yml:31]:
```suggestion
```
These tables are seeded and return on read, we seem to get 429 Too Many Requests before them but after a backoff the stream succeeds:

```
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing formulas"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceCoda runtimes:\nSyncing stream categories 0:00:06.010095\nSyncing stream docs 0:00:00.273884\nSyncing stream formulas 0:00:00.931787\nSyncing stream pages 0:00:00.947204\nSyncing stream permissions 0:00:00.752181\nSyncing stream tables 0:00:06.022509"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Backing off _send(...) for 5.0s (airbyte_cdk.sources.streams.http.exceptions.DefaultBackoffException: Request URL: https://coda.io/apis/v1/docs?limit=25, Response Code: 429, Response Text: {\"statusCode\":429,\"statusMessage\":\"Too Many Requests\",\"message\":\"Too Many Requests\"})"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Caught retryable error 'Request URL: https://coda.io/apis/v1/docs?limit=25, Response Code: 429, Response Text: {\"statusCode\":429,\"statusMessage\":\"Too Many Requests\",\"message\":\"Too Many Requests\"}' after 1 tries. Waiting 5 seconds then retrying..."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Successfully connected to stream controls, but got 0 records."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Marking stream controls as STARTED"}}
{"type": "TRACE", "trace": {"type": "STREAM_STATUS", "emitted_at": 1692983911302.9958, "stream_status": {"stream_descriptor": {"name": "controls", "namespace": null}, "status": "STARTED"}}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: controls "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Marking stream controls as RUNNING"}}
{"type": "TRACE", "trace": {"type": "STREAM_STATUS", "emitted_at": 1692983911932.323, "stream_status": {"stream_descriptor": {"name": "controls", "namespace": null}, "status": "RUNNING"}}}
{"type": "RECORD", "record": {"stream": "controls", "data": {"id": "ctrl-5Bcjb5X9g4", "type": "control", "href": "https://coda.io/apis/v1/docs/Mp5mvaMqF_/controls/ctrl-5Bcjb5X9g4", "name": "Clear template", "parent": {"id": "canvas-jtewKcxOL8", "type": "page", "href": "https://coda.io/apis/v1/docs/Mp5mvaMqF_/pages/canvas-jtewKcxOL8", "browserLink": "https://coda.io/d/_dMp5mvaMqF_/_suOL8", "name": "Tasks"}}, "emitted_at": 1692983911931}}
{"type": "RECORD", "record": {"stream": "controls", "data": {"id": "ctrl-QfxsEdOxRd", "type": "control", "href": "https://coda.io/apis/v1/docs/Mp5mvaMqF_/controls/ctrl-QfxsEdOxRd", "name": "Add task", "parent": {"id": "canvas-jtewKcxOL8", "type": "page", "href": "https://coda.io/apis/v1/docs/Mp5mvaMqF_/pages/canvas-jtewKcxOL8", "browserLink": "https://coda.io/d/_dMp5mvaMqF_/_suOL8", "name": "Tasks"}}, "emitted_at": 1692983911932}}
{"type": "RECORD", "record": {"stream": "controls", "data": {"id": "ctrl-DZnMLxTxU6", "type": "control", "href": "https://coda.io/apis/v1/docs/dStS3qWXwN/controls/ctrl-DZnMLxTxU6", "name": "Clear template", "parent": {"id": "canvas-df964o9xNM", "type": "page", "href": "https://coda.io/apis/v1/docs/dStS3qWXwN/pages/canvas-df964o9xNM", "browserLink": "https://coda.io/d/_ddStS3qWXwN/_suxNM", "name": "To-do list"}}, "emitted_at": 1692983912062}}
{"type": "RECORD", "record": {"stream": "controls", "data": {"id": "ctrl-HjNjJ6cj24", "type": "control", "href": "https://coda.io/apis/v1/docs/dStS3qWXwN/controls/ctrl-HjNjJ6cj24", "name": "Add task", "parent": {"id": "canvas-df964o9xNM", "type": "page", "href": "https://coda.io/apis/v1/docs/dStS3qWXwN/pages/canvas-df964o9xNM", "browserLink": "https://coda.io/d/_ddStS3qWXwN/_suxNM", "name": "To-do list"}}, "emitted_at": 1692983912062}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 4 records from controls stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Marking stream controls as STOPPED"}}
```

Comment (sh4sh) [reply airbyte-integrations/connectors/source-coda/acceptance-test-config.yml]: The tests for `controls`, `tables` and `rows` fail due to the initial 429

**Comment by sh4sh** [comment]
Nevermind, it looks like `rows` doesn't work in the original connector since we don't have any way of inferring what `tableIdOrName` to access. The user inputs `docId` in the connector configuration and will likely need to do the same for `tableId`, but since changing spec is a major change I think it should be its own PR.

File: airbyte-integrations/connectors/source-coda/acceptance-test-config.yml (hunk 1 above)
Comment (sh4sh) [reply airbyte-integrations/connectors/source-coda/acceptance-test-config.yml]: `rows` doesn't work with the existing connector configuration so I have removed my implementation

**Review by sh4sh — APPROVED** [review]

**Comment by sh4sh** [comment]
Thanks for your contribution

**Pull request merged**